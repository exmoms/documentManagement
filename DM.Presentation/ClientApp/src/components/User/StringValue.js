import React, { Component } from "react";
import TextField from "@material-ui/core/TextField";
import { withTranslation } from "react-i18next";

class StringValue extends Component {
  constructor(props) {
    super(props);
    this.state = { text: "" };
    this.handleValueChange = this.props.handleValueChange;
  }

  t = this.props["t"];

  handleChange = (event) => {
    let value = event.target.value;
    this.setState(
      { text: value },
      this.handleValueChange(value, this.props.label)
    );
  };

  render() {
    return (
      <div>
        <TextField
          id={this.props.label}
          label={this.t(this.props.label)}
          variant="outlined"
          onChange={this.handleChange}
          InputLabelProps={{
            shrink: true,
          }}
        />
      </div>
    );
  }
}

export default withTranslation()(StringValue);
