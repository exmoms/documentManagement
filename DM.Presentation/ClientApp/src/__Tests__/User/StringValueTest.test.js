import React from "react";
import ReactDom from "react-dom";
import StringValue from "../../components/User/StringValue";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { act } from "react-dom/test-utils";
import { mount } from "enzyme";
import TextField from "@material-ui/core/TextField";
afterEach(cleanup);
test("StringValue matches snapshot", () => {
  const component = renderer.create(<StringValue />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("StringValue", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<StringValue></StringValue>, div);
  });

  it("StringValue Functionality ", () => {
    let labelMock = "Text Value";
    let handleValueChangeMock = jest.fn();
    let attributeNameMock = "attrib1";
    let wrapper = mount(
      <StringValue
        label={labelMock}
        handleValueChange={handleValueChangeMock}
        attributeName={attributeNameMock}
      />
    );
    act(() => {
      wrapper
        .find(TextField)
        .at(0)
        .props()
        .onChange({ target: { value: "test2" } });
    });

    wrapper.update();

    expect(handleValueChangeMock).toHaveBeenCalledTimes(1);
    expect(handleValueChangeMock.mock.calls[0][0]).toStrictEqual("test2");
    expect(handleValueChangeMock.mock.calls[0][1]).toStrictEqual(labelMock);
  });
});
