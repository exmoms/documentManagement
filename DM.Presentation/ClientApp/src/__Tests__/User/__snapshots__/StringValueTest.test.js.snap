// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringValue matches snapshot 1`] = `
<div>
  <div
    className="MuiFormControl-root MuiTextField-root"
  >
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
      />
      <fieldset
        aria-hidden={true}
        className="PrivateNotchedOutline-root-39 MuiOutlinedInput-notchedOutline"
        style={
          Object {
            "paddingLeft": 8,
          }
        }
      >
        <legend
          className="PrivateNotchedOutline-legend-40"
          style={
            Object {
              "width": 0.01,
            }
          }
        >
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "&#8203;",
              }
            }
          />
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
